### PR TITLE
added pickle and yaml cache plugins

### PR DESCRIPTION
**Pull request by bcoca**
##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

pickle cache plugin

##### ANSIBLE VERSION

```
2.3
```

##### SUMMARY

yet another file based cache plugin

### CODE REVIEW

**Comment by mattclay** [comment]
CI failure due to PEP 8 issues:

```
2017-02-10 21:46:19 ERROR: PEP 8: lib/ansible/plugins/action/yaml.py: Does not exist. Remove from skip list (test/sanity/pep8/skip.txt).
2017-02-10 21:46:19 ERROR: PEP 8: lib/ansible/plugins/cache/yaml.py:86:161: E501 line too long (176 > 160 characters) (current)
```

**Comment by xen0l** [comment]
@bcoca have you considered using cPickle instead of pickle? It is faster and compatible (for this use case as far as I have read the source code)

**Comment by bcoca** [comment]
@xen0l now it prefers it

File: lib/ansible/plugins/cache/pickle.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+__metaclass__ = type
+
+import os
+import time
+import errno
+import codecs
+
+try:
+    import cPickle as pickle
+except ImportError
```
Comment (mattclay) [inline lib/ansible/plugins/cache/pickle.py]: CI failure due to syntax error (missing `:`).

File: test/sanity/pep8/skip.txt [hunk 2]
```diff
@@ -1 +1,3 @@
 lib/ansible/plugins/action/__init__.py
+lib/ansible/plugins/cache/pickle.py
+lib/ansible/plugins/cache/yaml.py
```
Comment (mattclay) [inline test/sanity/pep8/skip.txt:3]: Why are the PEP 8 checks being skipped for these files?

**Pull request closed without merging**